diff -urNp -x '*.orig' libasyncns-0.8.org/configure.ac libasyncns-0.8/configure.ac
--- libasyncns-0.8.org/configure.ac	2009-10-15 03:42:51.000000000 +0200
+++ libasyncns-0.8/configure.ac	2023-12-09 13:12:10.465786710 +0100
@@ -76,8 +76,8 @@ AC_CHECK_FUNCS([memset select strndup se
 
 AC_SYS_LARGEFILE
 
-AC_CHECK_LIB(nsl, gethostbyname)
-AC_CHECK_LIB(socket, connect)
+AC_SEARCH_LIBS(gethostbyname, nsl)
+AC_SEARCH_LIBS(connect, socket)
 
 ACF_SEARCH_FUNCTION([res_query], [-lresolv, -lbind],
   [], [AC_MSG_ERROR([res_query not found])],
